x-pack/plugins/painless_lab/public/plugin.tsx

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { Plugin, CoreStart, CoreSetup, PluginInitializerContext } from 'kibana/public';
+import { first } from 'rxjs/operators';
+import { EuiBetaBadge, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+
+import { FeatureCatalogueCategory } from '../../../../src/plugins/home/public';
+import { LICENSE_CHECK_STATE } from '../../licensing/public';
+
+import { PLUGIN } from '../common/constants';
+import { PluginDependencies } from './types';
+import { registerPainless } from './application/register_painless';
+
+export class PainlessLabUIPlugin implements Plugin<void, void, PluginDependencies> {
+  constructor(ctx: PluginInitializerContext) {}
+
+  async setup(
+    { http, getStartServices, uiSettings }: CoreSetup,
+    { devTools, home, licensing }: PluginDependencies
+  ) {
+    home.featureCatalogue.register({
+      id: PLUGIN.id,
+      title: i18n.translate('xpack.painlessLab.registryProviderTitle', {
+        defaultMessage: 'Painless Lab (beta)',
+      }),
+      description: i18n.translate('xpack.painlessLab.registryProviderDescription', {
+        defaultMessage: 'Simulate and debug painless code.',
+      }),
+      icon: '',
+      path: '/app/kibana#/dev_tools/painless_lab',
+      showOnHomePage: false,
+      category: FeatureCatalogueCategory.ADMIN,
+    });
+
+    devTools.register({
+      id: 'painless_lab',
+      order: 7,
+      title: (
+        <EuiFlexGroup gutterSize="s" alignItems="center">
+          <EuiFlexItem grow={false}>
+            {i18n.translate('xpack.painlessLab.displayName', {
+              defaultMessage: 'Painless Lab',
+            })}
+          </EuiFlexItem>
+
+          <EuiFlexItem grow={false} className="painlessLab__betaLabelContainer">
+            <EuiBetaBadge
+              label={i18n.translate('xpack.painlessLab.displayNameBetaLabel', {
+                defaultMessage: 'Beta',
+              })}
+              tooltipContent={i18n.translate('xpack.painlessLab.displayNameBetaTooltipText', {
+                defaultMessage: 'This feature might change drastically in future releases',
+              })}
+            />
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      ) as any,
+      enableRouting: false,
+      disabled: false,
+      // tooltipContent: xpackInfo.get('features.painlessLab.message'),
```

Comment:
I'm not sure what this is for or how to test (it was brought over from the previous code). @jloleysens or @cjcenizal - any ideas?

Reply:
Ah, I see. I'm going to delete this for now.

Currently, the app is redirecting back to /dev-tools if the license is invalid. This is [current functionality](https://github.com/elastic/kibana/blob/app/painless/x-pack/legacy/plugins/painless_lab/public/register.tsx#L64) that I brought over. It looks like this was originally copied over from `grokdebugger`, as I see it is doing the same thing.